detectable/src/main/java/com/synopsys/integration/detectable/detectables/pnpm/lockfile/process/PnpmLockYamlParserv6.java

```diff
@@ -0,0 +1,94 @@
+package com.synopsys.integration.detectable.detectables.pnpm.lockfile.process;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.collections4.MapUtils;
+import org.apache.commons.collections4.Predicate;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.Nullable;
+
+import com.synopsys.integration.detectable.detectable.codelocation.CodeLocation;
+import com.synopsys.integration.detectable.detectables.pnpm.lockfile.model.PnpmLockYamlv6;
+import com.synopsys.integration.detectable.detectables.pnpm.lockfile.model.PnpmProjectPackagev6;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.util.NameVersion;
+
+public class PnpmLockYamlParserv6 {
```

Comment:
Added a new parser as we have a lot of new v6 POJOs that have been loaded with the changes in the v6 lockfile. We need to obtain and deal with the information slightly differently and it made sense to separate this from v5. In general this PR differentiates v5 and v6 approaches by having different classes. This does not entangle the code and it allows for easy removal of the v5 code when the time comes.